tests/unit/customizations/s3/test_s3handler.py

```diff
@@ -827,6 +827,48 @@ def test_submit_move_adds_delete_source_subscriber(self):
         for i, actual_subscriber in enumerate(actual_subscribers):
             self.assertIsInstance(actual_subscriber, ref_subscribers[i])
 
+    def test_tagging_option_adds_set_tag_subscriber(self):
```

Comment:
It would be nice if we can get functional tests for this feature as it would serve a nice safety net in case we ever refactor all of these abstractions. I'd say we can do something similar to what we did for copy props functional tests: https://github.com/aws/aws-cli/blob/v2/tests/functional/s3/test_cp_command.py#L1349

And I say at the very least we should have a case for:
* When `--tagging` is provided for copies
* When `--tagging` is provided for uploads (both small and multipart uploads)
* When `--merge-tagging` and `--tagging` is provided